### PR TITLE
fix: format proto comments in Client Overview

### PR DESCRIPTION
**Pull request by alicejli**
Follow up PR to https://github.com/googleapis/sdk-platform-java/pull/2114 which breaks generation for java-bigqueryreservation due to errant formatting of the proto comments.

cl/587006892 contains the mock for bigqueryreservation's ReservationServiceClent.

I think it could make sense to add an integration test for this as the original integration tests didn't catch this bug, but I also understand that there are already a lot of integration tests. Thoughts?

### CODE REVIEW

File: gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java [hunk 1]
```diff
@@ -242,7 +242,7 @@ private static String createTableOfMethods(List<MethodAndVariants> methodAndVari
           .append(method.method)
           .append("</td>\n")
           .append("     <td>")
-          .append("<p>" + method.description + "</p>")
+          .append(CommentFormatter.formatAsJavaDocComment(method.description, null))
```
Comment (blakeli0) [inline gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java:245]: Can we make the whole table a String and format with `CommentFormatter`? This fix should be good for now, but in the future, someone else may add another line with dynamic content that could result in the same bug.

**Comment by blakeli0** [comment]
> I think it could make sense to add an integration test for this as the original integration tests didn't catch this bug, but I also understand that there are already a lot of integration tests. Thoughts?

Can we mock this scenario with showcase? By adding the problematic comment to [one of the showcase protos](https://github.com/googleapis/gapic-showcase/tree/3188c8a6089597d937ed6ab5b529716ab92d5ff3/schema/google/showcase/v1beta1).

File: gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java (hunk 1 above)
Comment (alicejli) [reply gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java:245]: The challenge with formatting the whole table with `CommentFormatter` is that it replaces *all* of the html elements for the table with the HTML-escaped character sequence for them, so the output doesn't stay formatted as a table. You can see the example output here: cl/572240700

**Review by burkedavison — APPROVED** [review]

**Comment by alicejli** [comment]
> > I think it could make sense to add an integration test for this as the original integration tests didn't catch this bug, but I also understand that there are already a lot of integration tests. Thoughts?
> 
> Can we mock this scenario with showcase? By adding the problematic comment to [one of the showcase protos](https://github.com/googleapis/gapic-showcase/tree/3188c8a6089597d937ed6ab5b529716ab92d5ff3/schema/google/showcase/v1beta1).

Good idea - PR opened: https://github.com/googleapis/gapic-showcase/pull/1391

**Comment by alicejli** [comment]
@blakeli0 I added a failing test for JavaWriter per our offline discussion. Let me know any additional thoughts, thanks!

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**